common/src/main/java/com/wynntils/services/mapdata/MapFeatureRenderer.java

```diff
@@ -130,4 +184,75 @@ public static void renderMapFeature(
 
         poseStack.popPose();
     }
+
+    public static boolean isHovered(
+            MapFeature mapFeature,
+            ResolvedMapAttributes attributes,
+            Vector2f mapCenter,
+            Vector2f screenCenter,
+            Vector2f rotationVector,
+            Vector2i mousePos,
+            float zoomRenderScale,
+            float zoomLevel,
+            float featureRenderScale) {
+        if (mapFeature instanceof MapLocation location) {
+            Location featureLocation = location.getLocation();
+            float dX = (featureLocation.x() - mapCenter.x()) * zoomRenderScale;
+            float dZ = (featureLocation.z() - mapCenter.y()) * zoomRenderScale;
+
+            float tempdX = dX * rotationVector.x() - dZ * rotationVector.y();
+            dZ = dX * rotationVector.y() + dZ * rotationVector.x();
+            dX = tempdX;
+
+            float renderX = screenCenter.x() + dX;
+            float renderZ = screenCenter.y() + dZ;
+
+            int labelHeight = (int) (FontRenderer.getInstance().getFont().lineHeight * featureRenderScale * TEXT_SCALE);
+
+            int yOffset = 0;
+
+            float iconAlpha = Services.MapData.calculateVisibility(attributes.iconVisibility(), zoomLevel);
+            Optional<MapIcon> icon = Services.MapData.getIcon(attributes.iconId());
+            boolean drawIcon = iconAlpha > MINIMUM_RENDER_ALPHA;
+            if (icon.isPresent() && drawIcon) {
```

Comment:
Feels like we are doing a lot over overlap here with the rendering. I wonder if we can express this logic only once... hmm....

Reply:
It's really hard to create such abstractions sadly. The other map features will be simpler to hover check.